index.bs

```diff
@@ -52,10 +70,192 @@ dictionary StorageBucketOptions {
+
+    1. Set |bucket|'s [=StorageBucket/quota value=] to |quota|.
+    
+    1. Set |shelf|'s [=bucket map=][|name|] to |bucket|.
+
+1. If |persisted| is true, set |bucket|'s [=/bucket mode=] to `"persistent"`.
+
+1. Set |bucket|'s [=StorageBucket/expiration value|expiration=] to |expires|.
+
+1. Let |storageBucket| be a new {{StorageBucket}}.
+
+1. Set |storageBucket|'s [=/storage bucket=] to |bucket|.
+
+1. Return |storageBucket|.
+
+</div>
+
+<div algorithm>
+
+To <dfn>validate a bucket name</dfn> given string |name|, run the following steps:
+
+1. If |name| contains any [=code point=] that is not [=ASCII lower alpha=], [=ASCII digit=], U+005F (_), or U+002D(-), then return failure.
+
+1. If |name| [=string/code point length=] is 0 or exceeds 64, then return failure.
+
+1. If |name| begins with U+005F (_) or U+002D(-), then return failure.
+
+1. Return.
+
+</div>
+
+When not null or undefined, {{StorageBucketOptions/durability}} is a hint to the user agent
+specifying the desired {{StorageBucket/durability}}. The user agent MAY
+create a new {{StorageBucket}} with this [=StorageBucket/durability value=]. The user agent
+MUST NOT modify the [=StorageBucket/durability value=] of an existing bucket.
+
 <h3 id="storage-bucket-delete">Deleting a bucket</h3>
+
+<div algorithm>
+
+The <dfn method for="StorageBucketManager">delete(|name|)</dfn> method steps are:
+
+1. Let |environment| be [=/this=]'s [=/relevant settings object=].
+
+1. Let |shelf| be the result of running [=obtain a local storage shelf=] given |environment|.
+
+1. If |shelf| is failure, then return [=a promise rejected with=] a {{TypeError}}.
+
+1. Let |p| be [=a new promise=].
+
+1. Run the following steps [=in parallel=]:
+
+    1. If the result of [=validate a bucket name=] with |name| is failure, then [=/reject=] |p| with an {{InvalidCharacterError}} and abort these steps.
+
+    1. Let |r| be the result of running [=remove a bucket=] with |shelf| and |name|.
+    
+    1. If |r| is failure, then [=/reject=] |p| with {{Type Error}}.
+
+    1. Otherwise, [=/resolve=] |p|.
+
+1. Return |p|.
+
+</div>
+    
+<div algorithm>
+
+To <dfn>remove a bucket</dfn> on a |shelf| given a bucket |name|, run the following steps:
+
+1. Let |bucket| be |shelf|’s [=bucket map=][|name|] if exists. Otherwise return.
+
+1. Remove [=map/key=] |name| in |shelf|'s [=bucket map=].
+
+1. Mark |bucket|'s [=storage bucket/removed=] as true.
+
+1. Return.
+
+<aside class="note">
+
+Specific storage endpoints may need to run additional actions to remove data when a storage bucket is marked as removed.
+
+</aside>
+
+Issue: [[IndexedDB-3]] needs to define how deletion occurs when data is evicted by quota.
+
+Issue: [[FS]] needs to define how deletion occurs for Origin Private File System when data is evicted by quota.
+
+Issue: [[Service-Workers]] needs to define how deletion occurs for CacheStorage when data is evicted by quota.
```

Comment:
It should also define how deletion works for service workers on eviction.

Reply:
Added to this issue